compiler/dyno/include/chpl/framework/ErrorBase.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef CHPL_FRAMEWORK_ERRORBASE_H
+#define CHPL_FRAMEWORK_ERRORBASE_H
+
+#include "chpl/framework/ErrorMessage.h"
+#include "chpl/uast/all-uast.h"
+#include "chpl/types/all-types.h"
+#include "llvm/ADT/Optional.h"
+
+namespace chpl {
+
+#define ERROR_CLASS(NAME, EINFO...) DIAGNOSTIC_CLASS(NAME, ERROR, EINFO)
+#define WARNING_CLASS(NAME, EINFO...) DIAGNOSTIC_CLASS(NAME, WARNING, EINFO)
+#define SYNTAX_CLASS(NAME, EINFO...) DIAGNOSTIC_CLASS(NAME, SYNTAX, EINFO)
+#define NOTE_CLASS(NAME, EINFO...) DIAGNOSTIC_CLASS(NAME, NOTE, EINFO)
+
+class ErrorWriter;
+
+enum ErrorType {
+  PARSE,
```

Comment:
Maybe put a comment above PARSE and GENERAL saying that they are included specifically here (rather than being in error-class-list.h) so that their behavior can be better customized to include different ways of constructing etc?